src/base/static/js/views/place-form-view.js

```diff
@@ -1,136 +1,259 @@
-
-      $('#datetimepicker').datetimepicker({ formatTime: 'g:i a' });
+      if (this.geometryEnabled) {
+        this.options.appView.hideCenterPoint();
+        this.options.appView.hideSpotlightMask();
+        this.geometryEditorView.render({
+          $el: this.$el,
+          iconUrl: this.formState.selectedCategoryConfig.icon_url
+        });
+      } else {
 
-      return this;
-    },
-    // called from the app view
-    postRender: function() {
-      // NOTE: the extra call to initialize the date-time picker is necessary here,
-      // because on a single-category form the call to initialize in the render() method
-      // above will fail, since the form content will not yet have been inserted into
-      // the DOM by the app view
-      if (this.formState.isSingleCategory) {
-        $('#datetimepicker').datetimepicker({ formatTime: 'g:i a' });
+        // In case the user switches from a geometry-enabled category
+        // to a geometry non-enabled category
+        this.geometryEditorView.tearDown();
+        this.options.appView.showNewPin();
       }
 
-      this.bindCategoryListeners();
+      this.initializeDatetimePicker();
+      this.initializeRichTextFields();
+      this.setUrlTitlePrefix();
     },
-    bindCategoryListeners: function() {
-      $(".category-btn-container").off().on("click", function(evt) {
-        $(this).prev().trigger("click");
+
+    setUrlTitlePrefix: function() {
+      var self = this, 
+      layer = _.find(this.options.mapConfig.layers, function(layer) {
+        return layer.id === self.formState.selectedCategoryConfig.dataset;
       });
+
+      this.$el.find(".url-prefix")
+        .html(
+          window.location.protocol + "//" + window.location.hostname + "/" +
+          (layer.useSlugForCustomUrls ? layer.slug + "/" : "")
+        );
+    },
+
+    showCategorySeparator: function() {
+      this.$("#category-btns hr").show();
+    },
+
+    hideCategorySeparator: function() {
+      this.$("#category-btns hr").hide();
+    },
+
+    hideSilhouettes: function() {
+      this.$(".place-form-silhouettes").hide();
     },
-    checkAutocomplete: function() {
-      var self = this,
-      storedValue;
 
-      this.formState.selectedCategoryConfig.fields.forEach(function(field, i) {
-        storedValue = Util.getAutocompleteValue(field.name);
-        self.formState.selectedCategoryConfig.fields[i].autocompleteValue = storedValue || null;
+    resetFormState: function() {
+      this.formState = {
+        selectedCategoryConfig: {
+          fields: []
+        },
+        attachmentData: [],
+        commonFormElements: this.options.placeConfig.common_form_elements || {}
+      }
+    },
+
+    determineRenderabilityForEachCategory: function() {
+      _.each(this.options.placeConfig.place_detail, function(place) {
+        _.extend(place, {
+          isAdmin: Util.getAdminStatus(place.dataset),
+          isRenderable: function() {
+            if (!place.includeOnForm) {
```

Comment:
I believe this is what you want: `place.includeOnForm && (!place.admin_only || Util.getAdminStatus(place.dataset))`. (Finally, endless boolean exercises in CSE 142 come in handy!)

Not saying you should write that in one line. I'd do:
```
var allowedToAccess = !place.admin_only || Util.getAdminStatus(place.dataset);
// (in situ)
place.includeOnForm && allowedToAccess
```

("Access" might be the wrong word--I don't know 100% what this is checking)